lib/config/configuration.go

```diff
@@ -483,21 +484,17 @@ func applyAuthConfig(fc *FileConfig, cfg *service.Config) error {
 	}
 	// read in and set authentication preferences
 	if fc.Auth.Authentication != nil {
-		authPreference, err := fc.Auth.Authentication.Parse()
+		cfg.Auth.Preference, err = fc.Auth.Authentication.Parse()
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		cfg.Auth.Preference = authPreference
 	}
-
-	var localAuth services.Bool
-	if fc.Auth.Authentication == nil || fc.Auth.Authentication.LocalAuth == nil {
-		localAuth = services.NewBool(true)
-	} else {
-		localAuth = *fc.Auth.Authentication.LocalAuth
+	if fc.Auth.DisconnectExpiredCert != nil {
+		cfg.Auth.Preference.SetOrigin(types.OriginConfigFile)
```

Comment:
Not with the current pathways since the default config including https://github.com/gravitational/teleport/blob/381bbb17ff4f2e3e1c36d23b207adb193a5ca979/lib/service/cfg.go#L854 would've already been applied by now.